confluent_kafka/schema_registry/config.py

```diff
@@ -0,0 +1,295 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Python 2 considers int an instance of str
+import json
+
+from .error import SchemaRegistryClientError
+
+try:
+    string_type = basestring  # noqa
+except NameError:
+    string_type = str
+
+from requests import utils, Session
+
+from ..config import Config
+from .schema_registry_client import SchemaRegistryClient
+from .serdes import SubjectNameStrategy, TopicRecordNameStrategy, TopicNameStrategy, \
+    RecordNameStrategy
+
+"""
+REST Client
+"""
+URL = 'url'
+REST_CLIENT = 'rest.client'
+"""
+Serializer Config
+"""
+SUBJECT_NAME_STRATEGY = 'subject.name.strategy'
+AUTO_REGISTRATION = 'auto.register.schemas'
+
+"""
+TLS configuration properties
+"""
+CA_LOCATION = 'ssl.ca.location'
+CERTIFICATE_LOCATION = 'ssl.certificate.location'
+KEY_LOCATION = 'ssl.key.location'
+
+"""
+Basic Auth configuration properties
+"""
+BASIC_AUTH_CREDENTIALS_SOURCE = 'basic.auth.credentials.source'
+BASIC_AUTH_USER_INFO = 'basic.auth.user.info'
+SASL_USERNAME = 'sasl.username'
+SASL_PASSWORD = 'sasl.password'
+SASL_MECHANISM = 'sasl.mechanism'
+
+
+class DefaultRestClient(Session):
```

Comment:
- Don't inherit from Session, use it internally 
- This doesn't really fit in config.py we can move it to the schema registry file wile keeping the schema registry oblivious to the actual implementation. Probably makes sense to have a base class as well to solidify the contract between the sr API and the RESTClient